sql/partition_info.cc

```diff
@@ -54,7 +54,7 @@ partition_info *partition_info::get_clone(bool reset /* = false */)
       mem_alloc_error(sizeof(partition_element));
       return NULL;
     }
-    memcpy(part_clone, part, sizeof(partition_element));
+    memcpy(static_cast<void*>(part_clone), part, sizeof(partition_element));
```

Comment:
This does not fix the root cause of the warning. If the actual fix is too complex for us at this point of 5.5 lifecycle, just patch the build for -Wno-class-memaccess instead.